### ISSUE TITLE
checkpoints: num(ber)/epoch awareness

### ISSUE THREAD

**Issue by casperdcl** [iterative/dvclive#113]
Solves two problems:

- `dvc exp run && dvc exp run` shouldn't re-run everything a second time
  + Yes, the user code can check this but so what? Same argument for non-checkpointed `dvc repro && dvc repro`. DVC should non re-run.
- interrupting `dvc exp run` (e.g. due to a runner timeout) and resuming shouldn't re-start from checkpoint zero.

----

- Part of https://github.com/iterative/cml/issues/560
- Possible solution https://github.com/iterative/dvclive/issues/68

**Comment by casperdcl** [comment]
not sure; `dvclive` documentation doesn't really exist but sounds like it would work

**Comment by dberenbaum** [comment]
Sorry @casperdcl, I was responding to @daavoo. I still think having `dvclive.range()` would be nice for a manual training loop.

> not sure; `dvclive` documentation doesn't really exist but sounds like it would work

Not for long! https://github.com/iterative/dvc.org/pull/2632

**Comment by casperdcl** [comment]
Ah I see. Well `for step in dvclive.range(...)` would make `dvclive.step` a bit redundant.

**Comment by dberenbaum** [comment]
To be more explicit, `dvclive.step` is for use with frameworks where the code looks like `model.fit(initial_epoch=dvclive.step, epochs=100)` instead of `for step in dvclive.range(100)`. There's no manual training loop, which I assumed was the concern raised by @daavoo.

**Comment by daavoo** [comment]
> `dvclive.step` is for use with frameworks where the code looks like `model.fit(initial_epoch=dvclive.step, epochs=100)` instead of `for step in dvclive.range(100): train(model)`. There's no manual training loop, which I assumed was the concern raised by @daavoo.

Yes, my concerns were about *ML Frameworks* where there is not an easy way to integrate the usage of `dvclive.step`, also was worried about the consistency between different `dvclive<>ML Framework` integrations given that not every *ML Framework* might have support for "resuming" or indicating an initial epoch that is not `0`.

> Ah I see. Well `for step in dvclive.range(...)` would make `dvclive.step` a bit redundant.

For some *ML Frameworks*, where the user writes it's own training loop, something like the code block bellow could be used instead of the hypothetical `dvclive.range` (unless we would want to wrap additional functionality under `dvclive.range`):

```
while dvclive.get_step() < 100:
    # train for one epoch
```

Anyhow, it looks that adding a new public method `dvclive.get_step()` would be trivial and has many potential usages, cc @dberenbaum
